File: s3/oop/Week12/CsApp/Counter.cs
public class Counter
{
    private int value;
    private int minValue;
    private int maxValue;

    public Counter(int minValue, int maxValue) {
        this.minValue = minValue;
        this.maxValue = maxValue;
        this.value = minValue;
    }

    public bool Increment() {
        this.value += 1;
        if (this.value > maxValue) {
            this.value = minValue;
            return true; // Counter rolled over
        }
        return false; // No rollover
    }

    public void Reset() {
        this.value = this.minValue;
    }

    public int Value {
        get { return value; }
    }

    public string GetPaddedValue() {
        return Value.ToString("D2");
    }
}

File: s3/oop/Week12/python/counter.py
class Counter:
  def __init__(self, min_value=0, max_value=59):
    self.min_value = min_value
    self.max_value = max_value
    self.value = min_value
  
  def increment(self): # Increment the counter.
    self.value += 1
    if self.value > self.max_value:
      self.value = self.min_value
      return True # Counter rolled over.
    return False # Counter did not roll over.
  
  def reset(self): # Reset the counter to the minimum value.
    self.value = self.min_value
    
  def __str__(self): # Get string value.
    return f"{self.value:02d}"